core/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -355,6 +355,9 @@ protected Query blendTermQuery(Term term, MappedFieldType fieldType) {
                     if (query instanceof FuzzyQuery) {
                         QueryParsers.setRewriteMethod((FuzzyQuery) query, fuzzyRewriteMethod);
                     }
+                    if (query == null) {
```

Comment:
I don't follow how the query can be null here. I thought that either we throw exception or the query is not null?